src/main/java/com/aws/greengrass/localdebugconsole/SimpleHttpServer.java

```diff
@@ -154,13 +218,54 @@ public void startup() throws InterruptedException {
         reportState(State.RUNNING);
     }
 
+    static String fingerprintCert(X509Certificate cert)
+            throws NoSuchAlgorithmException, CertificateEncodingException {
+        StringBuilder sb = new StringBuilder();
+        for (byte b : MessageDigest.getInstance("SHA-256").digest(cert.getEncoded())) {
+            sb.append(String.format("%02X ", b));
+        }
+        return sb.toString().trim();
+    }
+
+    private static X509Certificate selfSign(KeyPair keyPair, String subjectDN) throws OperatorCreationException,
+            CertificateException,
+            IOException
+    {
+        java.security.Provider bcProvider = new BouncyCastleProvider();
+        Security.addProvider(bcProvider);
+
+        long now = System.currentTimeMillis() - TimeUnit.SECONDS.toMillis(60);
+        Date startDate = new Date(now);
+
+        X500Name dnName = new X500Name("cn=" + subjectDN);
+        BigInteger certSerialNumber = new BigInteger(Long.toString(now)); // Using the current timestamp as the certificate serial number
+
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(startDate);
+        calendar.add(Calendar.YEAR, 1); // 1 Year validity
```

Comment:
we self sign every start up - so why do you need to set the end date as 1 year?

Is this chosen just to avoid having to regenerate?

Reply:
Just if it stays running for a long time I don't want it to run out easily. There's no reason to have it be even shorter since we're only really doing this to add encryption to the channel and not as much for AuthN.

Reply:
Bumped it to 10 years.